eravm-stable-interface/src/lib.rs

```diff
@@ -0,0 +1,136 @@
+use u256::{H160, U256};
+
+pub trait StateInterface {
+    fn read_register(&self, register: u8) -> (U256, bool);
+    fn set_register(&mut self, register: u8, value: U256, is_pointer: bool);
+
+    fn number_of_callframes(&self) -> usize;
+    /// zero is the current frame, one is the frame before that etc.
+    fn callframe(&mut self, n: usize) -> impl CallframeInterface + '_;
+
+    fn read_heap_byte(&self, heap: HeapId, index: u32) -> u8;
+    fn write_heap_byte(&mut self, heap: HeapId, index: u32, byte: u8);
+
+    fn flags(&self) -> Flags;
+    fn set_flags(&mut self, flags: Flags);
+
+    fn transaction_number(&self) -> u16;
+    fn set_transaction_number(&mut self, value: u16);
+
+    fn context_u128_register(&self) -> u128;
+    fn set_context_u128_register(&mut self, value: u128);
+
+    /// Returns the new value and the amount of pubdata paid for it.
```

Comment:
Is this comment outdated? Or does it belong to `get_storage()`? In general, I'd suggest defining a `struct` for the returned value, so that the purpose of fields is obvious from their name.